editor/editor/mode/visual.js

```diff
@@ -20,11 +20,25 @@ function Blocks( { blocks, onChange } ) {
 	return (
 		<div className="editor-mode-visual">
 			<div>
-				{ blocks.map( ( block, index ) =>
-					<div key={ index }>
-						{ wp.blocks.getBlockSettings( block.blockType ).edit( block.attributes, onChangeBlock( index ) ) }
-					</div>
-				) }
```

Comment:
If we fallback to a different blockType as suggested above, all the changes here are not needed anymore.